src/Std/Internal/Async/Basic.lean

```diff
@@ -109,6 +319,458 @@ def getState (x : AsyncTask α) : BaseIO IO.TaskState :=
 
 end AsyncTask
 
+/--
+A `EMaybeTask ε α` represents a computation that either:
+
+- Is immediately available as an `α` value, or
+- Is an asynchronous computation that will eventually produce an `α` value or an error `ε`
+-/
+inductive EMaybeTask (ε : Type) (α : Type)
+  | pure : α → EMaybeTask ε α
+  | ofTask : ETask ε α → EMaybeTask ε α
```

Comment:
It is unclear to me why you have specialized the notion of `MaybeTask` to `EMaybeTask`. Why not define the more general `MaybeTask` and implement `EMaybeTask` as `MaybeTask (Except ε α)` (paralleling `Task` and `ETask`)?

Reply:
In my implementation of `MaybeTask (Except ε α)` with `BaseIO` instead, the generated IR was like 2 times bigger and a little bit worse, I was trying to optimize based on the generated code and it was my result :S. I just focused on using the exception mechanism inside the `EIO`.